src/components/calc.js

```diff
@@ -1,60 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Display from './calc/display';
 import ButtonsContainer from './calc/buttonsContainer';
 import calculate from './calc/logic/calculate';
 import './calc.css';
 
-class Calc extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: null,
-      next: null,
-      operation: null,
-    };
-    this.handleClick = this.handleClick.bind(this);
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-  }
+const Calc = () => {
+  const [calc, setCalc] = useState({ total: null, next: null, operation: null });
 
-  handleClick = (e) => {
-    this.updateState(e.target.dataset.name);
+  const updateState = async (obj, key) => {
+    let { total, next, operation } = await calculate(obj, key);
```

Comment:
* I get errors in the console when i perform division by zero operation (ie 8 / 0) kindly fix this error 🙏 
![image](https://user-images.githubusercontent.com/52497246/129133699-4c90b3bc-349b-4902-a695-9b324eb428d0.png)

Reply:
You got me! 😆 I'll fix this and I'll try to implement a max length for inputs aswell so numbers don't flow out of the box.